emu_sv/state_vector.py

```diff
@@ -102,14 +124,7 @@ def inner(self, other: State) -> float | complex:
         Returns:
             the inner product
         """
-        assert isinstance(
-            other, StateVector
-        ), "Other state also needs to be a StateVector"
-        assert (
-            self.vector.shape == other.vector.shape
-        ), "States do not have the same number of sites"
-
-        return torch.vdot(self.vector, other.vector).item()
+        return self._inner(other).cpu().item()
```

Comment:
this should return a `torch.Tensor` too, so it's essentially what you have in `_inner` now.